lib/devicetrust/native/device_windows.go

```diff
@@ -511,6 +528,161 @@ func solveTPMEnrollChallenge(
+	// Clear up the results of any previous credential activation
+	if err := os.Remove(todoCredentialActivationPath); err != nil {
+		err := trace.ConvertSystemError(err)
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err, "clearing previous credential activation results")
+		}
+	}
+
+	// Assemble the parameter list. We encoded any binary data in base64.
+	// The list is then converted to a space delimited string.
+	// These parameters cause `tsh` to invoke HandleActivateCredential.
+	params := strings.Join([]string{
+		"-d", // enable debug mode so relevant logs appear in the new window
+		"device",
+		"activate-credential",
+		"--encrypted-credential",
+		base64.StdEncoding.EncodeToString(encryptedCredential.Credential),
+		"--encrypted-credential-secret",
+		base64.StdEncoding.EncodeToString(encryptedCredential.Secret),
+	}, " ")
+
+	// Create pointers to invoke the windows syscall with.
+	operation := "runas"
+	lpOperation, err := syscall.UTF16PtrFromString(operation)
+	if err != nil {
+		return nil, trace.Wrap(err, "converting operation to ptr")
+	}
+	lpFile, err := syscall.UTF16PtrFromString(exe)
+	if err != nil {
+		return nil, trace.Wrap(err, "converting exe to ptr")
+	}
+	lpParameters, err := syscall.UTF16PtrFromString(params)
+	if err != nil {
+		return nil, trace.Wrap(err, "converting params to ptr")
+	}
+	lpDirectory, err := syscall.UTF16PtrFromString(cwd)
+	if err != nil {
+		return nil, trace.Wrap(err, "converting cwd to ptr")
+	}
+
+	log.Debug("Starting elevated process.")
+	// https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutew
+	err = windows.ShellExecute(
+		0, // hwnd
+		lpOperation,
+		lpFile,
+		lpParameters,
+		lpDirectory,
+		windows.SW_NORMAL,
+	)
+	// The windows.ShellExecute blocks until the user answers the UAC dialogue.
+	// If they reject the UAC dialogue, an error is returned:
+	//   syscall.Errno The operation was canceled by the user.
+	if err != nil {
+		return nil, trace.Wrap(err, "invoking ShellExecuteW")
+	}
+
+	// Ensure we clean up the results of the execution once we are done with
+	// it.
+	defer func() {
+		if err := os.Remove(todoCredentialActivationPath); err != nil {
+			log.WithError(err).Debug("Failed to clean up credential activation result")
+		}
+	}()
+
+	log.Debug("Waiting for credential activation solution from elevated process.")
+	// Wait for result from child process, polling every half second for up to
+	// 10 seconds.
+	for i := 0; i < 20; i++ {
+		// TODO: Context cancellation ?
+		log.Debug("Checking for credential activation solution from elevated process.")
+		solutionBytes, err := os.ReadFile(todoCredentialActivationPath)
+		if err == nil {
+			return solutionBytes, nil
+
+		}
+		log.WithError(err).Debug("Error when polling for solution. Waiting to try again.")
+		time.Sleep(time.Millisecond * 500)
+	}
+	// TODO: Return last error that occurred
+	return nil, trace.Errorf("failed to receive credential activation results from elevated process")
```

Comment:
```suggestion
	return nil, trace.BadParameter("failed to receive credential activation results from elevated process")
```

I've been told to avoid naked Errorf before.